sql/expression/function/strcmp_test.go

```diff
@@ -36,10 +37,13 @@ func TestStrCmp(t *testing.T) {
 		{"equal strings", sql.Text, sql.Text, sql.NewRow("a", "a"), int(0), nil},
```

Comment:
I think there's a few more tests cases we should add...

The first group is testing automatic type coercion:
- `select strcmp(1, 1);` => `0`
-  `select strcmp(1, 0);` => `1`
-  `select strcmp(0, 1);` => `-1`
-  `select strcmp(42, "42");` => `0`
-  `select strcmp(DATE("2022-01-01"), 0);` => `1`
-  `select strcmp(DATE("2022-01-01"), "2022-01-01");` => `0`

The other group to flesh out more is tests with collation:
- `select strcmp("abc" collate utf8mb4_0900_ai_ci, "ABC" collate utf8mb4_0900_ai_ci);` => `0`
- `select strcmp("abc" collate utf8mb4_0900_bin, "ABC" collate utf8mb4_0900_bin);` => `1`

@Hydrocharged – I got the same results as MySQL for the collation tests above, but I hit something odd with the `utf8mb4_0900_as_cs` collation... The `strcmp` test above works correctly with `utf8mb4_0900_bin` (returns `1` on both Dolt and MySQL), but when I change to use `utf8mb4_0900_as_cs`, MySQL returns `-1` and Dolt is still returning `1`. Since Stephanie's code doesn't directly work with collations (just calls out to `StringType.Compare`), I'm wondering if there's something busted with our configuration for the `utf8mb4_0900_as_cs` collation?

Reply:
So the issue with `utf8mb4_0900_as_cs` is due to it not even being implemented 🤣
https://github.com/dolthub/go-mysql-server/tree/main/sql/encodings
It’s supposed to throw an error saying it isn’t implemented, so that’s an issue!

Reply:
mystery solved!! thanks for confirming @Hydrocharged! 😁